src/test/java/org/openrefine/extensions/commons/importer/FileFetcherTest.java

```diff
@@ -0,0 +1,84 @@
+package org.openrefine.extensions.commons.importer;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import okhttp3.HttpUrl;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+
+public class FileFetcherTest {
+
+    /**
+     * Test row generation from mocked api calls and paging with a cmcontinue token
+     */
+    @Test
+    public void testNext() throws Exception {
+
+        try (MockWebServer server = new MockWebServer()) {
+            server.start();
+            HttpUrl url = server.url("/w/api.php");
+            String jsonResponseContinue = "{\"batchcomplete\":\"\",\"continue\":{\"cmcontinue\":"
+                    + "\"file|4492e4a5047|13935\",\"continue\":\"-||\"},\"query\":{\"categorymembers\":"
+                    + "[{\"pageid\":127722,\"ns\":6,\"title\":\"File:LasTres.jpg\",\"type\":\"file\"},"
+                    + "{\"pageid\":127752,\"ns\":6,\"title\":\"File:Pejiballes.jpg\",\"type\":\"file\"}]}}";
+            server.enqueue(new MockResponse().setBody(jsonResponseContinue));
+            String jsonResponse = "{\"batchcomplete\":\"\",\"query\":{\"categorymembers\":"
+                    + "[{\"pageid\":112928,\"ns\":6,\"title\":\"File:Esferas de CR.jpg\",\"type\":\"file\"},"
+                    + "{\"pageid\":112933,\"ns\":6,\"title\":\"File:Playa Gandoca.jpg\",\"type\":\"file\"}]}}";
+            server.enqueue(new MockResponse().setBody(jsonResponse));
+            List<String> category = Collections.singletonList("Category:Costa Rica");
+            FileFetcher fileFetcher = new FileFetcher(url.toString(), category.get(0));
+
+            List<Object> rows = new ArrayList<>();
+            rows.add(fileFetcher.next());
```

Comment:
It's easiest to test `hasNext` in conjunction with `next` so I would suggest you add assertions to check `hasNext` in this test already :)